telegram/ext/conversationhandler.py

```diff
@@ -423,6 +444,37 @@ def _get_key(self, update: Update) -> Tuple[int, ...]:
 
         return tuple(key)
 
+    def _resolve_promise(self, state: Tuple) -> object:
+        old_state, new_state = state
+        try:
+            res = new_state.result(0)
+            res = res if res is not None else old_state
+        except Exception as exc:
+            self.logger.exception("Promise function raised exception")
+            self.logger.exception("%s", exc)
+            res = old_state
+        finally:
+            if res is None and old_state is None:
+                res = self.END
+        return res
+
+    def _schedule_job(
+        self,
+        new_state: object,
+        dispatcher: 'Dispatcher',
+        update: Update,
+        context: Optional[CallbackContext],
+        conversation_key: Tuple[int, ...],
+    ) -> None:
+        if new_state != self.END:
+            self.timeout_jobs[
+                conversation_key
+            ] = dispatcher.job_queue.run_once(  # type:ignore[union-attr]
```

Comment:
We should add some try-except here and in the except clause either call `dispatcher.dispatch_error` or a least log something to warning or debug level about the timeout job not being set. This is relevant when the `done_callback` is actually run only in `Promise.run`, where we have no means of directly dispatching the error.

This case should also be tested, probably my mocking `run_once` to raise an excption.